NullPointer Exception when using fn:replace()

### Description:
APIM 3.1.0 copies the Saxon-HE-9.5.1-8.jar to the '<HOME>/libs/endorsed' directory but this directory is no longer used. This results in xalan-2.7.1 library being used to compile _fn:replace()_ function instead of Saxon-HE-9.5.1-8 causing a null pointer exceptions because the function is only supported in Saxon.

### Steps to reproduce:

1. Add below synapse property using wso2am-3.1.0/repository/conf/deployment.toml as below (to enable the failover for XPath 2.0)
```
 [synapse_properties]
 'synapse.xpath.dom.failover.enabled'="true"
```
2. Start the server and create a rest API with below in-flow mediation
```
 <sequence xmlns="http://ws.apache.org/ns/synapse" name="In">
     <property xmlns:fn="http://www.w3.org/2005/xpath-functions" name="NewLocation" expression="fn:replace('xxxxxxxciam', 'ciam', 'api')" scope="default" type="STRING"/>
 </sequence>
```
3. Invoke the API
4. You can observe below null pointer exception

```
ERROR - SynapseXPath Evaluation of the XPath expression fn:replace('xxxxxxxciam', 'ciam', 'api') resulted in an error
javax.xml.xpath.XPathExpressionException: javax.xml.transform.TransformerException: java.lang.NullPointerException
	at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(XPathExpressionImpl.java:189) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(XPathExpressionImpl.java:221) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.synapse.util.xpath.SynapseXPath.evaluateDOMXPath(SynapseXPath.java:655) ~[synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.util.xpath.SynapseXPath.stringValueOf(SynapseXPath.java:488) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.mediators.builtin.PropertyMediator.getResultValue(PropertyMediator.java:374) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.mediators.builtin.PropertyMediator.mediate(PropertyMediator.java:112) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.wso2.carbon.apimgt.gateway.handlers.ext.APIManagerExtensionHandler.mediate_aroundBody0(APIManagerExtensionHandler.java:72) [org.wso2.carbon.apimgt.gateway_6.6.163.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.ext.APIManagerExtensionHandler.mediate(APIManagerExtensionHandler.java:57) [org.wso2.carbon.apimgt.gateway_6.6.163.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.ext.APIManagerExtensionHandler.handleRequest_aroundBody2(APIManagerExtensionHandler.java:89) [org.wso2.carbon.apimgt.gateway_6.6.163.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.ext.APIManagerExtensionHandler.handleRequest(APIManagerExtensionHandler.java:78) [org.wso2.carbon.apimgt.gateway_6.6.163.jar:?]
	at org.apache.synapse.rest.API.process(API.java:367) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.rest.RESTRequestHandler.apiProcessNonDefaultStrategy(RESTRequestHandler.java:149) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.rest.RESTRequestHandler.dispatchToAPI(RESTRequestHandler.java:95) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.rest.RESTRequestHandler.process(RESTRequestHandler.java:71) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:327) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.synapse.core.axis2.SynapseMessageReceiver.receive(SynapseMessageReceiver.java:98) [synapse-core_2.1.7.wso2v147.jar:2.1.7-wso2v147]
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180) [axis2_1.6.1.wso2v41.jar:?]
	at org.apache.synapse.transport.passthru.ServerWorker.processNonEntityEnclosingRESTHandler(ServerWorker.java:368) [synapse-nhttp-transport_2.1.7.wso2v147.jar:?]
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:189) [synapse-nhttp-transport_2.1.7.wso2v147.jar:?]
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172) [axis2_1.6.1.wso2v41.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: javax.xml.transform.TransformerException: java.lang.NullPointerException
	at org.apache.xpath.jaxp.JAXPExtensionsProvider.extFunction(JAXPExtensionsProvider.java:204) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.functions.FuncExtFunction.execute(FuncExtFunction.java:208) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.XPath.execute(XPath.java:337) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.XPath.execute(XPath.java:303) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.eval(XPathExpressionImpl.java:121) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.eval(XPathExpressionImpl.java:91) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(XPathExpressionImpl.java:176) ~[xalan-2.7.2.wso2v2.jar:?]
	... 25 more
Caused by: java.lang.NullPointerException
	at org.apache.xpath.jaxp.JAXPExtensionsProvider.extFunction(JAXPExtensionsProvider.java:197) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.functions.FuncExtFunction.execute(FuncExtFunction.java:208) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.XPath.execute(XPath.java:337) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.XPath.execute(XPath.java:303) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.eval(XPathExpressionImpl.java:121) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.eval(XPathExpressionImpl.java:91) ~[xalan-2.7.2.wso2v2.jar:?]
	at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(XPathExpressionImpl.java:176) ~[xalan-2.7.2.wso2v2.jar:?]
	... 25 more
```

--- a/modules/distribution/product/src/main/startup-scripts/wso2server.sh
+++ b/modules/distribution/product/src/main/startup-scripts/wso2server.sh
@@ -263,6 +263,10 @@ for t in "$CARBON_HOME"/lib/*.jar
 do
     CARBON_CLASSPATH="$CARBON_CLASSPATH":$t
 done
+for t in "$CARBON_HOME"/lib/endorsed/*.jar
+do
+    CARBON_CLASSPATH="$CARBON_CLASSPATH":$t
+done
 
 
 
@@ -348,6 +352,7 @@ do
     -DworkerNode=false \
     -DenableCorrelationLogs=false \
     -Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
+    -Djavax.xml.xpath.XPathFactory:http://java.sun.com/jaxp/xpath/dom=net.sf.saxon.xpath.XPathFactoryImpl \
     org.wso2.carbon.bootstrap.Bootstrap $*
     status=$?
 done

